### PR TITLE
Feature/apl 694 p2p download

### PR DESCRIPTION
**Pull request by oleksiylukin**
Yet more error situation to process but we need deploy to tn2 to debug

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/FileDownloader.java [hunk 1]
```diff
@@ -17,48 +14,90 @@
...
-    
-    public FileDownloader() {      
+
+    ExecutorService executor;
+    List<Future<Boolean>> runningDownloaders = new ArrayList<>();
+
+    @Inject
+    public FileDownloader(DownloadableFilesManager manager) {
+        this.manager = manager;
+        this.executor = Executors.newFixedThreadPool(DOWNLOAD_THREADS);
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/FileDownloader.java]: Add @PreDestroy method for executor shutdown

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/db/ShardDataSourceCreateHelperTest.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ void getShardName() {
         createHelper = new ShardDataSourceCreateHelper(extension.getDatabaseManager(), 1L);
         createHelper.createUninitializedDataSource();
         String shardName = createHelper.getShardName();
-        assertEquals("apl-blockchain-shard-1", shardName);
+        assertEquals("apl-blockchain-shard-1-chain-b5d7b697-f359-4ce5-a619-fa34b6fb01a5", shardName);
```
Comment (AndrewBoyarsky) [inline apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/db/ShardDataSourceCreateHelperTest.java]: Maybe use shorten form?

Comment (alukin) [reply apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/db/ShardDataSourceCreateHelperTest.java]: No, thanks

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/FileDownloader.java (hunk 1 above)
Comment (alukin) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/FileDownloader.java]: Thanks, it is imortant. fixed.

**Pull request merged**